### ISSUE TITLE
Hardened Kernel Config File for Virtual Machines (VMs) ("cloud kernel") and Hosts

### ISSUE THREAD

**Issue by adrelanos** [clipos/bugs#38]
A kernel config specialized for better security inside virtual machines is in development.

The development preview version can be found here:
https://github.com/Whonix/hardened-kernel/blob/master/usr/share/hardened-kernel/hardened-vm-kernel

This work is being done by @madaidan who also contributed pull requests to [linux-hardened](https://github.com/anthraxx/linux-hardened).

https://github.com/anthraxx/linux-hardened/pulls?utf8=%E2%9C%93&q=author%3Amadaidan

Discussions about the kernel config happen mostly in Whonix forums.

https://forums.whonix.org/t/kernel-recompilation-for-better-hardening/7598/214

The hardened kernel config was contributed by @madaidan to the @Whonix project but as the maintainer of Whonix I think that it is not the most suitable project to maintain a kernel config. It would be more impactful and would get more eyes on it if it was hosted here.

Therefore I am wondering if there is any chance you would accept a pull request for a hardened VM config file? Which folder would be suitable for such a config file?

@madaidan is also working on a hardened bare metal (i.e. non-VM) kernel config:
https://github.com/Whonix/hardened-kernel/blob/master/usr/share/hardened-kernel/hardened-host-kernel

**Comment by msalaun-anssi** [comment]
To complete @tsautereau-anssi's [comment](https://github.com/a13xp0p0v/kconfig-hardened-check/issues/25#issuecomment-573735007), the kernel configuration is split in sets and selected with the [clipos-kernel_compute_configuration()](https://github.com/clipos/src_portage_clipos/blob/d69aac258c9bd4cf907fa9f38c348e42cae16edb/eclass/clipos-kernel.eclass#L107) helper.

**Comment by madaidan** [comment]
Where are the sets located? is it https://github.com/clipos/src_platform_config-linux-hardware/tree/master/kernel_config?

**Comment by tsautereau-anssi** [comment]
> Where are the sets located? is it https://github.com/clipos/src_platform_config-linux-hardware/tree/master/kernel_config?

Yes, exactly. And as pointed out by @msalaun-anssi, these sets are then handled at build-time by the [clipos-kernel_compute_configuration()](https://github.com/clipos/src_portage_clipos/blob/d69aac258c9bd4cf907fa9f38c348e42cae16edb/eclass/clipos-kernel.eclass#L107) helper, which itself calls our [make-config.sh](https://github.com/clipos/src_platform_config-linux-hardware/blob/992d5b40336612ec65d2d5c458da36e3a0ff0ba0/make-config.sh) script.

**Comment by msalaun-anssi** [comment]
The documentation for the kernel config is available here: https://docs.clip-os.org/clipos/kernel.html

**Comment by madaidan** [comment]
Would you also accept kernel patches or should I just send them to linux-hardened?

**Comment by tsautereau-anssi** [comment]
We do accept kernel patches. However, if you think your patches can benefit other people, it's probably a better idea to at least try to get them merged into linux-hardened first.

**Comment by madaidan** [comment]
Alright, I'll just keep sending them to linux-hardened then. I've also created a few kernel config/sysctl changes https://github.com/clipos/src_platform_config-linux-hardware/pulls
